drivers/stepper/adi_tmc/tmc2130.c

```diff
@@ -0,0 +1,300 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 Navimatix GmbH
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT adi_tmc2130
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/counter.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/drivers/stepper.h>
+#include "../step_dir/step_dir_stepper_common.h"
+#include "tmc2130_reg.h"
+#include "adi_tmc_spi.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(tmc2130, CONFIG_STEPPER_LOG_LEVEL);
+
+struct tmc2130_config {
+	struct step_dir_stepper_common_config common;
+	struct gpio_dt_spec en_pin;
+	struct spi_dt_spec spi;
+	bool stealth_chop_enabled;
```

Comment:
```suggestion
	const uint32_t gconf;
```
it's better to store the entire gconf register rather than a single parameter. If certain parameters for gconf are not available in adi,trinamic-gconf, please add them over there.

Reply:
could you give a second look at how we initialize gconf in tmc5xxx series :)
https://github.com/zephyrproject-rtos/zephyr/blob/2016f05860c457b428a6c18eedd0806e40a4a431/drivers/stepper/adi_tmc/tmc50xx.c#L745

Reply:
Yeah but I am asking you add the remaining flags in adi,trinamic-gconf.yaml, as the contributor of tmc51xx did as well and then write to the gconf register as done in tmc5xxx series, we are already following a pattern.

I just checked gconf has I think 18 bits in your case, so you can either use bitfield or uint32_t, it's upto, I personally don't mind.

Reply:
The gconf register has a macro for its creation, which is currently sufficient.
Concerning future extensibility, nothing in the current implementation makes it difficult to replace the current bool value with the full register in the future if it becomes necessary.

`stealth_chop_enabled` is already in `adi,trinamic-gconf.yaml` as `en-pwm-mode,` i just missed, either because it was added after I started working on the driver or because the flag name has nothing todo with stealth_chop. I will delete my own binding for stealth_chop.